### DISCUSSION TITLE
OnTouched checks for every field

### DISCUSSION THREAD

**Discussion by theprantadutta** [react-hook-form/react-hook-form#3934] [Q&A]
Hi, Thanks for the library. I really love it. But I came across a problem. See I am using useForm with onTouched mode, now when I touch a field it validates all fields like if I am typing the name, it validates email too. Well, it doesn't actually show the error message but as I am using a test function to send a request to an API to check if the email is unique or not, I am seeing it's sending a lot of request to the API, and I don't want that. I just want to validate the field which is touched.

I don't think it's a bug, that's why I am creating a discussion, not an issue. Thanks in advance.

**Comment by bluebill1049** [comment]
>  it validates all fields like if I am typing the name

can you share a codesadnbox?

**Reply by theprantadutta** [reply]
Sorry, code sandbox is showing some weird errors today. Anyway here's my code:
this is the yup validation. Here's I am checking whether the email is unique or not.
```
  import Yup from "../lib/yup";
  import { api} from "../utils/api";
  
  export const registerValidationSchema = Yup.object({
    name: Yup.string().required("Required"),
    email: Yup.string()
      .email("Invalid email")
      .test("Unique Email", "Email already been taken", function (value) {
        return new Promise(async (resolve, _) => {
          try {
            await api().post("/unique-email", {
              email: value,
            });
            resolve(false);
          } catch (e) {
            resolve(true);
          }
        });
      })
      .required("Required"),
    password: Yup.string()
      .min(6, "Password should be at least six letters")
      .required("Required"),
    password_confirmation: Yup.string()
      .oneOf([Yup.ref("password")], "Passwords must match")
      .required("Required"),
  });

```
This is my register file
```
  const {
      register,
      handleSubmit,
      errors,
      setError,
    } = useForm<RegisterFormValues>({
      resolver: yupResolver(registerValidationSchema),
      mode: "onTouched",
      reValidateMode: "onSubmit",
    });
```
Now I am declaring my fields to be validated on touch, right? But when I am writing a name or password, it validates the email too. 
See I am sending a request to the API every time now. I just want to send it when the email field is touched.
I really hope you got it.

**Reply by bluebill1049** [reply]
This is because validation is going run against the entire schema, as far as i know yup doesn't support field-level validation, if they do we can fix that in the next major version as we provide better context to `resolver` in the v7.

**Reply by theprantadutta** [reply]
So should I use the react hook form's validate method here?

**Reply by bluebill1049** [reply]
what about just add that logic to `onBlur` method?

**Reply by bluebill1049** [reply]
build validator works, but i don't want you to change everything that you already got working.

**Reply by theprantadutta** [reply]
I mean it works to an extent. Now when I write on a field, then it sends the request. That will work...thanks bill :)

**Reply by bluebill1049** [reply]
np.
